source/common/http/http1/conn_pool.cc

```diff
@@ -90,9 +90,10 @@ void ConnPoolImpl::createNewConnection() {
 
 ConnectionPool::Cancellable* ConnPoolImpl::newStream(StreamDecoder& response_decoder,
                                                      ConnectionPool::Callbacks& callbacks) {
-  host_->cluster().stats().upstream_rq_total_.inc();
-  host_->stats().rq_total_.inc();
   if (!ready_clients_.empty()) {
+    host_->cluster().stats().upstream_rq_total_.inc();
```

Comment:
Can you potentially move this into `attachRequestToCleint()` and remove from below? I think it's better to have the logic in one place and it's more accurate since we still may not send the request upstream if it gets cancelled while pending.

Reply:
This makes sense to me, thanks for catching that pending request case.